interface/repository/pokemon_repository.go

```diff
@@ -3,30 +3,93 @@ package repository
 import (
 	"os"
 
+	resty "github.com/go-resty/resty/v2"
 	"github.com/gocarina/gocsv"
 	"github.com/luispaulin/api-challenge/domain/model"
 )
 
 type pokemonRepository struct {
-	file *os.File
+	file   *os.File
+	client *resty.Client
 }
 
 // PokemonRepository for interface
 type PokemonRepository interface {
 	FindAll(pokemons []*model.Pokemon) ([]*model.Pokemon, error)
+	Sync() (string, int, error)
 }
 
 // NewPokemonRepository for interface
-func NewPokemonRepository(db *os.File) PokemonRepository {
-	return &pokemonRepository{db}
+func NewPokemonRepository(db *os.File, client *resty.Client) PokemonRepository {
+	return &pokemonRepository{db, client}
 }
 
 func (pr *pokemonRepository) FindAll(pokemons []*model.Pokemon) ([]*model.Pokemon, error) {
 
+	// Set reader t file's beginning
+	if _, err := pr.file.Seek(0, 0); err != nil {
+		return nil, err
+	}
+
+	// Check if empty file
+	fileInfo, err := pr.file.Stat()
+	if err != nil {
+		return nil, err
+	}
+	if fileInfo.Size() == 0 {
+		return nil, nil
+	}
+
 	// Parses csv file info to pokemon slice
 	if err := gocsv.UnmarshalFile(pr.file, &pokemons); err != nil {
 		return nil, err
 	}
 
 	return pokemons, nil
 }
+
+func (pr *pokemonRepository) Sync() (string, int, error) {
+
+	var pokemons []*model.Pokemon
+	// TODO this struct here?
+	type Response struct {
```

Comment:
If the struct is only used in one package, it could be declared in this package.
But do it after the imports and not inside of this function.

Reply:
Ok, done.